packages/oracle/src/dialect.ts

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved
+
+import type { Sequelize } from '@sequelize/core';
+import { AbstractDialect } from '@sequelize/core';
+import type { SupportableNumericOptions } from '@sequelize/core/_non-semver-use-at-your-own-risk_/abstract-dialect/dialect.js';
+import { createSpecifiedOrderedBindCollector } from '@sequelize/core/_non-semver-use-at-your-own-risk_/utils/sql.js';
+import { EMPTY_ARRAY } from '@sequelize/utils';
+import { CONNECTION_OPTION_NAMES } from './_internal/connection-options.js';
+import * as DataTypes from './_internal/data-types-overrides';
+import { OracleConnectionManager } from './connection-manager';
+import type { OracleConnectionOptions } from './connection-manager.js';
+import { OracleQueryGenerator } from './query-generator.js';
+import { OracleQueryInterface } from './query-interface.js';
+import { OracleQuery } from './query.js';
+
+export interface OracleDialectOptions {}
+
+const numericOptions: SupportableNumericOptions = {
+  zerofill: false,
+  unsigned: true,
+};
+
+export class OracleDialect extends AbstractDialect<OracleDialectOptions, OracleConnectionOptions> {
+  static readonly supports = AbstractDialect.extendSupport({
+    'VALUES ()': true,
+    'LIMIT ON UPDATE': true,
+    lock: false,
+    forShare: 'LOCK IN SHARE MODE',
+    index: {
+      collate: false,
+      length: false,
+      parser: false,
+      type: false,
+      using: false,
+    },
+    constraints: {
+      restrict: false,
+      onUpdate: false,
+    },
+    returnValues: false,
+    returnIntoValues: true,
+    'ORDER NULLS': true,
+    schemas: true,
+    inserts: {
+      updateOnDuplicate: false,
+    },
+    indexViaAlter: false,
+    dataTypes: {
+      COLLATE_BINARY: true,
+      GEOMETRY: false,
+      JSON: true,
+      INTS: numericOptions,
+      DOUBLE: numericOptions,
+      DECIMAL: { unconstrained: true },
+      TIME: {
+        precision: false,
+      },
+    },
+    jsonOperations: true,
+    jsonExtraction: {
+      quoted: true,
+    },
+    dropTable: {
+      cascade: true,
+    },
+    renameTable: {
+      changeSchema: false,
+    },
+    delete: {
+      limit: true,
+    },
+    startTransaction: {
+      useBegin: true,
+    },
+    upserts: true,
+    bulkDefault: true,
+    topLevelOrderByRequired: true,
+  });
+
+  readonly connectionManager: OracleConnectionManager;
+  readonly queryGenerator: OracleQueryGenerator;
+  readonly queryInterface: OracleQueryInterface;
+  readonly Query = OracleQuery;
+
+  constructor(sequelize: Sequelize, options: OracleDialectOptions) {
+    super({
+      dataTypesDocumentationUrl:
+        'https://docs.oracle.com/en/database/oracle/oracle-database/23/sqlrf/Data-Types.html',
+      minimumDatabaseVersion: '18.0.0',
```

Comment:
We're testing on 18.4.0. Should we make that the minimum database version?

Reply:
I tried changing docker image to 18.4.0, It gives some issue. I think some explicit settings of VM RAM are needed. Hence skipped it.